compiler/env/OMRCPU.hpp

```diff
@@ -169,21 +163,21 @@ class OMR_EXTENSIBLE CPU
     * @param[in] p : the input processor type
     * @return true when current processor is the same as the input processor type
     */
-   bool is(OMRProcessorArchitecture p) const { return _processorDescription.processor == p; }
+   bool is(OMRProcessorArchitecture p);
```

Comment:
Why can't we do `const` any longer? Similar comment for other APIs.

Reply:
I think eventually we can do `const`..
Current issue with const is this:
```cpp
/home/vsts/work/1/s/compiler/x/env/OMRCPU.cpp:258:19: error: passing ‘const OMR::X86::CPU’ as ‘this’ argument discards qualifiers [-fpermissive]
       return self()->is_old_api(p);
                   ^
In file included from /home/vsts/work/1/s/compiler/x/env/OMRCPU.hpp:37:0,
                 from /home/vsts/work/1/s/compiler/env/CPU.hpp:25,
                 from /home/vsts/work/1/s/compiler/x/env/OMRCPU.cpp:24:
/home/vsts/work/1/s/compiler/env/OMRCPU.hpp:110:13: note:   in call to ‘TR::CPU* OMR::CPU::self()’
    TR::CPU *self();
             ^
/home/vsts/work/1/s/compiler/x/env/OMRCPU.cpp:259:15: error: passing ‘const OMR::X86::CPU’ as ‘this’ argument discards qualifiers [-fpermissive]
    TR_ASSERT(self()->is_test(p), "old api and new api did not match, processor %d", p);
               ^
```

When `self()` is used inside a const function the compiler complains, I got around it by removing the `const`. I'm not entirely sure why this happens (with self() call, we can't ensure the constness or something?).
When we remove "old" apis, these `self()` calls will be removed as well and this will no longer be the problem.

Are you okay will removing `const` for now? unless you know how to work around this

Reply:
Yeah that is fine. I was hoping it was a simple oversight and adding it back is trivial, but it appears this is not the case. We can just let it go for now.

Reply:
Issue opened to track leftover (https://github.com/eclipse/omr/issues/5247)